com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/handler/ThreadsRequestHandler.java

```diff
@@ -114,6 +126,53 @@
         return CompletableFuture.completedFuture(response);
     }
 
+    private CompletableFuture<Response> resumeAll(Requests.ThreadOperationArguments arguments, Response response, IDebugAdapterContext context) {
+        context.getExceptionManager().removeAllExceptions();
+        context.getDebugSession().resume();
+        context.getProtocolServer().sendEvent(new Events.ContinuedEvent(arguments.threadId, true));
+        context.getRecyclableIdPool().removeAllObjects();
+        return CompletableFuture.completedFuture(response);
+    }
+
+    private CompletableFuture<Response> resumeOthers(Requests.ThreadOperationArguments arguments, Response response, IDebugAdapterContext context) {
+        List<ThreadReference> threads = DebugUtility.getAllThreadsSafely(context.getDebugSession());
+        for (ThreadReference thread : threads) {
+            long threadId = thread.uniqueID();
+            if (threadId != arguments.threadId && thread.isSuspended()) {
+                context.getExceptionManager().removeException(threadId);
+                DebugUtility.resumeThread(thread);
+                context.getProtocolServer().sendEvent(new Events.ContinuedEvent(threadId));
+                checkThreadRunningAndRecycleIds(thread, context);
+            }
+        }
+
+        return CompletableFuture.completedFuture(response);
+    }
+
+    private CompletableFuture<Response> pauseAll(Requests.ThreadOperationArguments arguments, Response response, IDebugAdapterContext context) {
+        context.getDebugSession().suspend();
+        context.getProtocolServer().sendEvent(new Events.StoppedEvent("pause", arguments.threadId, true));
+        return CompletableFuture.completedFuture(response);
+    }
+
+    private CompletableFuture<Response> pauseOthers(Requests.ThreadOperationArguments arguments, Response response, IDebugAdapterContext context) {
+        List<ThreadReference> threads = DebugUtility.getAllThreadsSafely(context.getDebugSession());
+        for (ThreadReference thread : threads) {
+            long threadId = thread.uniqueID();
+            if (threadId != arguments.threadId && !thread.isSuspended()) {
+                try {
+                    thread.suspend();
+                    context.getProtocolServer().sendEvent(new Events.StoppedEvent("pause", threadId));
+                } catch (ObjectCollectedException ex) {
+                    // ObjectCollectionException can be thrown if the thread has already completed (exited) in the VM,
```

Comment:
before the try-catch block, can we use `ThreadReference.status()` to determine if it has completed?

Reply:
fixed.